addons/web_editor/tools.py

```diff
@@ -0,0 +1,133 @@
+# -*- encoding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import re
+import requests
+
+from markupsafe import Markup
+
+from odoo import _
+from odoo.tools import image_process
+from odoo.modules.module import get_module_resource
+
+
+# To detect if we have a valid URL or not
+valid_url_regex = r'^(http:\/\/|https:\/\/|\/\/)[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$'
+
+# Regex for few of the widely used video hosting services
+player_regexes = {
+    'youtube': r'^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube(-nocookie)?\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((?:\w|-){11})\S*$',
+    'vimeo': r'\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*',
+    'dailymotion': r'(?:.+dailymotion.com\/(video|hub|embed\/video|embed)|dai\.ly)\/([^_?]+)[^#]*(#video=([^_&]+))?',
+    'instagram': r'(?:(.*)instagram.com|instagr\.am)\/p\/(.[a-zA-Z0-9-_\.]*)',
+    'youku': r'(.*).youku\.com\/(v_show\/id_|embed\/)(.+)',
+}
+
+
+def get_video_source_data(video_url):
+    ''' Computes the valid source, document ID and regex match from given URL
+        (or False in case of invalid URL).
+    '''
+    if not video_url:
+        return False
+
+    if re.search(valid_url_regex, video_url):
+        youtube_match = re.search(player_regexes['youtube'], video_url)
+        if youtube_match:
+            return ('youtube', youtube_match.groups()[1], youtube_match)
```

Comment:
```suggestion
            return ('youtube', youtube_match[1], youtube_match)
```
match objects are indexable (also applies to the rest of the method)

Reply:
with group indexes starting at 0 though oops :grimacing: